src/ax/saga/workflow.go

```diff
@@ -22,30 +22,33 @@ type Workflow struct {
 
 // NewWorkflow returns a saga that forwards to the given aggregate.
 //
-// Workflows are a specialization of sagas that handle events and produce
-// commands.
+// Workflows are a specialization of sagas that handle commands and/or events
+// and produce commands.
 //
 // It accepts a prototype data instance which is cloned for new instances.
 //
-// For each event type to be handled, the aggregate must implement a "handler"
+// For each message type to be handled, the aggregate must implement a "handler"
 // method that adheres to one of the following signatures:
 //
-//     func (ev *<T>, ax.CommandExecutor)
-//     func (ev *<T>, env ax.Envelope, ax.CommandExecutor)
+//     func (m *<T>, ax.CommandExecutor)
+//     func (m *<T>, env ax.Envelope, ax.CommandExecutor)
 //
-// Where T is a struct type that implements ax.Event.
+// Where T is a struct type that implements ax.Message.
 //
 // Handler methods are responsible for mutating the state of the workflow and
-// producing new commands, based on the event being handled. They may inspect
-// the current state of the workflow, and then return zero or more commands
-// to be executed.
+// producing new commands, based on the command or event being handled. They may
+// inspect the current state of the workflow, and then return zero or more
+// commands to be executed.
 //
-// The names of handler methods are meaningful to the workflow system. If an
-// event is meant to trigger a new workflow instance, its handler method's name
-// must begin with "StartWhen". Other handler methods must begin with "When". By
+// The names of handler methods are meaningful to the workflow system. If a
+// command or event is meant to trigger a new workflow instance, its handler
+// method's name must begin with "StartWhen". Other handler methods must begin
+// with "When". By
 // convention these prefixes are followed by the message name, such as:
 //
+//     func (*BankTransferWorkflow) StartWhenTransferStarted(*messages.StartTransfer, ax.CommandExecutor)
 //     func (*BankTransferWorkflow) StartWhenTransferStarted(*messages.TransferStarted, ax.CommandExecutor)
+//     func (*BankTransferWorkflow) WhenAccountDebited(*messages.DebitAccount, ax.CommandExecutor)
 //     func (*BankTransferWorkflow) WhenAccountDebited(*messages.AccountDebited, ax.CommandExecutor)
```

Comment:
Hrm, I think these prefixes need to be revised.  So far throughout Ax, the word "when" is handler methods always implies an event.  In `Aggregate` for example, "event appliers" start with `When`, but command handlers do not require any specific naming.

We could keep `StartWhen` and `When` for events, and come up with some other prefixes for commands, but I'm not sure what they might be.  Any thoughts?

Reply:
Perhaps, it's still worth to differentiate signatures by command and event and not to use any prefixes for command signatures similar to aggregates?

Reply:
Yeah I think that might be best.  Aggregates don't currently require a prefix for commands, but I'm happy to add one there to keep consistency between aggregates and workflows.

/cc @koden-km